daemons/fenced/fenced_remote.c

```diff
@@ -1367,6 +1368,22 @@ find_best_peer(const char *device, remote_fencing_op_t * op, enum find_best_peer
     return NULL;
 }
 
+static gboolean
+is_device_support_on_action(const remote_fencing_op_t * op, const peer_device_info_t *peer, const char *device)
+{
+    if (peer != NULL && pcmk__str_eq(op->action, "on", pcmk__str_casei)) {
+        device_properties_t *props = g_hash_table_lookup(peer->devices, device);
+
+        if (props) {
+            if (!pcmk_is_set(props->flags, st_device_supports_on)) {
+                return FALSE;
+            }
+        }
```

Comment:
I'm thinking that this could be done earlier, in find_peer_device(), where we already have props.

That would prevent find_best_peer() -> grab_peer_device() fro setting props->executed if we're going to ignore the device, and get_op_total_timeout() would automatically skip the device.

That leaves just one case, in find_best_peer() when topology is not used (where the "Simple fencing" log message is). So, we could call this function there.

With that approach, instead of calling this function after calling find_best_peer(), find_best_peer() will have already returned NULL in this situation, so nothing more needs to be done.

What do you think?

Reply:
HI Ken,

Thanks for your comment.

Possibly, I think that this change can be realized simply by changing the processing of find_peer_device().
I will check and test it.

Many thanks,
Hideo Yamauchi.